### PR TITLE
fix(DTFS2-6024): BSS date validation

### PR DESCRIPTION
**Pull request by Zainzzkk**
# Issue:
* BSS allowed for the year to be 2 numbers (eg 22 instead of 2022) which caused some functions to assume the year 0022

# Changes made:
* Added validation using joi to check length is 4 and only digits entered for the number
* Updated tests

### CODE REVIEW

File: portal-api/src/v1/validation/bond-rules/facility-stage-issued-rules/requested-cover-start-date.js [hunk 1]
```diff
@@ -28,6 +29,10 @@ module.exports = (submittedValues, deal, errorList) => {
     const nowDate = moment().startOf('day');
 
     if (!dealHasBeenSubmitted) {
+      // validates the coverStartDateYear is 4 digits long and only numbers and returns error in validation if not
+      const schema = Joi.string().length(4).pattern(/^[0-9]+$/).required();
+      const validation = schema.validate(requestedCoverStartDateYear);
+
       if (moment(requestedCoverStartDateTimestamp).isBefore(nowDate)) {
         newErrorList.requestedCoverStartDate = {
           text: 'Requested Cover Start Date must be on the application submission date or in the future',
```
[the remark this answers is not in the document]
Comment (Zainzzkk) [reply portal-api/src/v1/validation/bond-rules/facility-stage-issued-rules/requested-cover-start-date.js:38]: I am not sure, I think we should keep it as is

File: portal-api/src/v1/validation/loan-rules/facility-stage-unconditional/requested-cover-start-date.js [hunk 2]
```diff
@@ -26,6 +27,9 @@ module.exports = (submittedValues, deal, errorList) => {
 
   if (requestedCoverStartDateTimestamp) {
     const nowDate = moment().startOf('day');
+    // validates the coverStartDateYear is 4 digits long and only numbers and returns error in validation if not
+    const schema = Joi.string().length(4).pattern(/^[0-9]+$/).required();
+    const validation = schema.validate(requestedCoverStartDateYear);
 
     if (!dealHasBeenSubmitted) {
       if (moment(requestedCoverStartDateTimestamp).isBefore(nowDate)) {
```
[the remark this answers is not in the document]
Comment (Zainzzkk) [reply portal-api/src/v1/validation/loan-rules/facility-stage-unconditional/requested-cover-start-date.js:35]: see above

File: portal-api/src/v1/validation/bond-rules/facility-stage-issued-rules/requested-cover-start-date.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (Zainzzkk) [reply portal-api/src/v1/validation/bond-rules/facility-stage-issued-rules/requested-cover-start-date.js:38]: already have

File: portal-api/src/v1/validation/loan-rules/facility-stage-unconditional/requested-cover-start-date.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (Zainzzkk) [reply portal-api/src/v1/validation/loan-rules/facility-stage-unconditional/requested-cover-start-date.js:35]: already have

**Review by abhi-markan — APPROVED** [review]
Good work

**Pull request merged**